File: PS Multi Tools/Classes/NPSPKG.vb
Imports System.ComponentModel
Imports System.Runtime.CompilerServices

Public Class NPSPKG

    Implements INotifyPropertyChanged

    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    Protected Sub NotifyPropertyChanged(<CallerMemberName> Optional propertyName As String = "")
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
    End Sub

    Private _PackageName As String
    Private _PackageDescription As String
    Private _PackageURL As String
    Private _PackageContentID As String
    Private _PackageRAP As String
    Private _PackageRegion As String
    Private _PackageDate As String
    Private _PackageDLCs As String
    Private _PackageSize As String
    Private _IsSelected As Boolean
    Private _PackagezRIF As String
    Private _PackageReqFW As String
    Private _PackageTitleID As String
    Private _GameCoverSource As ImageSource
    Private _PackageCoverSource As String

    Public Property PackageName As String
        Get
            Return _PackageName
        End Get
        Set
            _PackageName = Value
        End Set
    End Property

    Public Property PackageDescription As String
        Get
            Return _PackageDescription
        End Get
        Set
            _PackageDescription = Value
        End Set
    End Property

    Public Property PackageURL As String
        Get
            Return _PackageURL
        End Get
        Set
            _PackageURL = Value
        End Set
    End Property

    Public Property PackageTitleID As String
        Get
            Return _PackageTitleID
        End Get
        Set
            _PackageTitleID = Value
        End Set
    End Property

    Public Property PackageContentID As String
        Get
            Return _PackageContentID
        End Get
        Set
            _PackageContentID = Value
        End Set
    End Property

    Public Property PackageRAP As String
        Get
            Return _PackageRAP
        End Get
        Set
            _PackageRAP = Value
        End Set
    End Property

    Public Property PackagezRIF As String
        Get
            Return _PackagezRIF
        End Get
        Set
            _PackagezRIF = Value
        End Set
    End Property

    Public Property PackageReqFW As String
        Get
            Return _PackageReqFW
        End Get
        Set
            _PackageReqFW = Value
        End Set
    End Property

    Public Property PackageRegion As String
        Get
            Return _PackageRegion
        End Get
        Set
            _PackageRegion = Value
        End Set
    End Property

    Public Property PackageDate As String
        Get
            Return _PackageDate
        End Get
        Set
            _PackageDate = Value
        End Set
    End Property

    Public Property PackageDLCs As String
        Get
            Return _PackageDLCs
        End Get
        Set
            _PackageDLCs = Value
        End Set
    End Property

    Public Property PackageSize As String
        Get
            Return _PackageSize
        End Get
        Set
            _PackageSize = Value
        End Set
    End Property

    Public Property IsSelected As Boolean
        Get
            Return _IsSelected
        End Get
        Set
            _IsSelected = Value
        End Set
    End Property

    Public Property PackageCoverSource As String
        Get
            Return _PackageCoverSource
        End Get
        Set
            _PackageCoverSource = Value
        End Set
    End Property

    Public Property GameCoverSource As ImageSource
        Get
            Return _GameCoverSource
        End Get
        Set
            _GameCoverSource = Value
            NotifyPropertyChanged("GameCoverSource")
        End Set

    End Property

End Class

File: PSMultiTools/Classes/NPSPKG.cs
using IronSoftware.Drawing;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace PSMultiTools.Classes
{

    public class NPSPKG : INotifyPropertyChanged
    {
        private string? _packageName;
        private string? _packageDescription;
        private string? _packageURL;
        private string? _packageTitleID;
        private string? _packageContentID;
        private string? _packageRAP;
        private string? _packagezRIF;
        private string? _packageReqFW;
        private string? _packageRegion;
        private string? _packageDate;
        private string? _packageDLCs;
        private string? _packageSize;
        private bool _isSelected;
        private string? _packageCoverSource;
        private AnyBitmap? _gameCoverSource;

        public string? PackageName
        {
            get => _packageName;
            set => SetProperty(ref _packageName, value);
        }

        public string? PackageDescription
        {
            get => _packageDescription;
            set => SetProperty(ref _packageDescription, value);
        }

        public string? PackageURL
        {
            get => _packageURL;
            set => SetProperty(ref _packageURL, value);
        }

        public string? PackageTitleID
        {
            get => _packageTitleID;
            set => SetProperty(ref _packageTitleID, value);
        }

        public string? PackageContentID
        {
            get => _packageContentID;
            set => SetProperty(ref _packageContentID, value);
        }

        public string? PackageRAP
        {
            get => _packageRAP;
            set => SetProperty(ref _packageRAP, value);
        }

        public string? PackagezRIF
        {
            get => _packagezRIF;
            set => SetProperty(ref _packagezRIF, value);
        }

        public string? PackageReqFW
        {
            get => _packageReqFW;
            set => SetProperty(ref _packageReqFW, value);
        }

        public string? PackageRegion
        {
            get => _packageRegion;
            set => SetProperty(ref _packageRegion, value);
        }

        public string? PackageDate
        {
            get => _packageDate;
            set => SetProperty(ref _packageDate, value);
        }

        public string? PackageDLCs
        {
            get => _packageDLCs;
            set => SetProperty(ref _packageDLCs, value);
        }

        public string? PackageSize
        {
            get => _packageSize;
            set => SetProperty(ref _packageSize, value);
        }

        public bool IsSelected
        {
            get => _isSelected;
            set => SetProperty(ref _isSelected, value);
        }

        public string? PackageCoverSource
        {
            get => _packageCoverSource;
            set => SetProperty(ref _packageCoverSource, value);
        }

        public AnyBitmap? GameCoverSource
        {
            get => _gameCoverSource;
            set => SetProperty(ref _gameCoverSource, value);
        }

        public event PropertyChangedEventHandler? PropertyChanged;

        protected bool SetProperty<T>(ref T field, T value, [CallerMemberName] string? propertyName = null)
        {
            if (EqualityComparer<T>.Default.Equals(field, value))
                return false;

            field = value;
            OnPropertyChanged(propertyName);
            return true;
        }

        protected void OnPropertyChanged([CallerMemberName] string? propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}